bin/src/worker.rs

```diff
@@ -62,13 +62,23 @@ pub fn start_workers(executable_path: String, config: &Config) -> anyhow::Result
         let mut worker = Worker::new(index as u32, pid, command_channel, scm_socket, config);
 
         // the new worker expects a status message at startup
-        if let Some(command_channel) = worker.worker_channel.as_mut() {
-            command_channel.blocking();
-            command_channel.write_message(&ProxyRequest {
-                id: format!("start-status-{}", index),
-                order: ProxyRequestOrder::Status,
-            });
-            command_channel.nonblocking();
+        if let Some(worker_channel) = worker.worker_channel.as_mut() {
+            // DISCUSS: should we propagate the error instead of printing it?
+            if let Err(e) = worker_channel.blocking() {
```

Comment:
Should be tested

Reply:
We've got a load of tests, actually! They test the creationn and use of blocking and nonblocking channels, and the `blocking()` method as well.